File: 128-longest-consecutive-sequence/main.go
func longestConsecutive(nums []int) int {
    
    numsSet := make(map[int]bool)
    for _, num := range nums {
        numsSet[num] = true
    }

    var longest int = 0
    for num := range numsSet {
            var length int = 1
            for numsSet[num + length] {
                length++
            }
            longest = max(longest, length)            
        }
    }

    return longest
}


func max(a, b int) int {
    if a < b {
        return b
    }
    

    return a
}

File: 128-longest-consecutive-sequence/main.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {

        std::unordered_set<int> nums_set(nums.begin(),  nums.end());
        int longest = 0;

        for (int num: nums_set) {
            if ( nums_set.find(num - 1) == nums_set.end()) {
                int length = 1;
                while (nums_set.find(num + length) != nums_set.end()) {
                    length++;
                }

                longest = std::max(longest, length);
            }
        }

        return longest;
    }
};
